lib/logstash/outputs/kafka.rb

```diff
@@ -275,13 +275,16 @@ def retrying_send(batch)
       break if failures.empty?
 
       # Otherwise, retry with any failed transmissions
-      batch = failures
-      delay = @retry_backoff_ms / 1000.0
-      logger.info("Sending batch to Kafka failed. Will retry after a delay.", :batch_size => batch.size,
-                  :failures => failures.size, :sleep => delay);
-      sleep(delay)
+      if remaining != nil && remaining < 0
```

Comment:
it seems that this condition is not useful here, IMO if we write the condition as:
```
if remaining.nil? || remaining >= 0
  delay = @retry_backoff_ms / 1000.0
  logger.info("Sending batch to Kafka failed. Will retry after a delay.", :batch_size => batch.size, :failures => failures.size, :sleep => delay)
  batch = failures
  sleep(delay)
end
```
and let the initial condition at the beginning of the method would handle the failure case?

Reply:
👍